### PR TITLE
Fan show

### PR DESCRIPTION
**Pull request by DelphineHelie**

### CODE REVIEW

**Comment by victorlacombe** [comment]
pareil pour les visuels avant de pusher please :)

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,2 +1,18 @@
 class CommentsController < ApplicationController
+  def index
+    @comments = Comment.all
+  end
+
+  def find
+    @fan = Fan.find(params[:id])
+    comments = Comment.where(fan_id: @fan)
```
Comment (victorlacombe) [inline app/controllers/comments_controller.rb]: remplacer comment par @comments

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,2 +1,18 @@
 class CommentsController < ApplicationController
```
Comment (victorlacombe) [inline app/controllers/comments_controller.rb]: Add authorizations in comments policy + in methods

File: app/controllers/fans_controller.rb [hunk 3]
```diff
@@ -14,8 +14,10 @@ def index
   def show
```
Comment (victorlacombe) [inline app/controllers/fans_controller.rb]: Vérifier les authorizations (fan policy) + les authorize

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -1,2 +1,18 @@
...
+  def index
+    @comments = Comment.all
+  end
+
+  def find
+    @fan = Fan.find(params[:id])
+    comments = Comment.where(fan_id: @fan)
+    @comment = Comment.find(params[:id])
+    videos = Video.where(video_id_youtube: @comment)
```
Comment (victorlacombe) [inline app/controllers/comments_controller.rb]: authorize @comment
@videos = policy_scope(Video).where(...)
dans video policy scope --> resolve.all
@eliotteliott

File: app/controllers/memos_controller.rb [hunk 5]
```diff
@@ -15,7 +15,8 @@ def update
     @memo = Memo.find(params[:id])
     authorize @memo
     @memo.update(memo_params)
-    redirect_to edit_memo_path(@memo)
+    @fan = Fan.find(params[:id])
```
Comment (victorlacombe) [inline app/controllers/memos_controller.rb]: authorize @fan

File: app/controllers/video_controller.rb [hunk 6]
```diff
@@ -1,2 +1,4 @@
 class VideoController < ApplicationController
-end
+  def index
+    @videos = Video.all
```
Comment (victorlacombe) [inline app/controllers/video_controller.rb]: idem pour le policy scope de video

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (eliotteliott) [reply app/controllers/comments_controller.rb]: done

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (eliotteliott) [reply app/controllers/comments_controller.rb]: done

File: app/controllers/comments_controller.rb (hunk 4 above)
Comment (eliotteliott) [reply app/controllers/comments_controller.rb]: ok 

class VideoPolicy < ApplicationPolicy
  class Scope < Scope
    def resolve
      scope.all

File: app/controllers/memos_controller.rb (hunk 5 above)
Comment (eliotteliott) [reply app/controllers/memos_controller.rb]: authorize @fan done

File: app/controllers/fans_controller.rb (hunk 3 above)
Comment (eliotteliott) [reply app/controllers/fans_controller.rb]: done

**Pull request merged**